src/app/routes/onboarding-gate.tsx

```diff
@@ -20,10 +21,15 @@ interface OnboardingGateProps {
 export function OnboardingGate({ children }: OnboardingGateProps) {
   const keyDetails = useCurrentKeyDetails();
   const currentInMemoryKey = useDefaultWalletSecretKey();
+  const isLedger = useHasLedgerKeys();
 
   if (
-    keyDetails?.type === 'software' &&
-    hasAlreadyMadeWalletAndPlaintextKeyInMemory(keyDetails.encryptedSecretKey, currentInMemoryKey)
+    (keyDetails?.type === 'software' &&
```

Comment:
This works but it's very tricky to read with somewhat of a ternary and function call inside of an `if`. 

I'd go for a small refactor to make it easier to understand

```
const isLedger = useHasLedgerKeys();
const hasKeyInMemory = hasAlreadyMadeWalletAndPlaintextKeyInMemory(
        keyDetails.encryptedSecretKey,
        currentInMemoryKey
      );
const canGoHome = (keyDetails?.type === 'software' && hasKeyInMemory) || isLedger;

  if (canGoHome){
...
```

For now lets leave it as is, working well but in future I think small refactors like that are worth it

Reply:
completely agree here, just was in a hurry a bit

Reply:
yeah thats fine, and understandable! nice job getting it fixed so quickly 👍